### PR TITLE
Spacewalk cherry-pick 2018-08-25

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

Cherry pick interresting commits from spacewalk project

### CODE REVIEW

**Comment by mcalmer** [comment]
@renner - I hope syntax errors are gone now and changelogs are added as well

**Comment by mcalmer** [comment]
gitarro re-run java_pgsql_tests !!!

**Comment by mcalmer** [comment]
gitarro re-run java_lint_checkstyle !!!

**Comment by mcalmer** [comment]
@renner - I need a review

File: client/debian/apt-spacewalk/apt-spacewalk.spec [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+%endif
+
+%if %{undefined python2_sitelib}
+%global python2_sitelib %(%{__python2} -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())")
+%endif
+
+Name: apt-spacewalk
+Summary: Spacewalk plugin for Advanced Packaging Tool
+%if %{_vendor} == "debbuild"
+Packager: Spacewalk Project <spacewalk-devel@redhat.com>
```
Comment (Conan-Kudo) [inline client/debian/apt-spacewalk/apt-spacewalk.spec]: This probably needs to be adjusted to point to the Uyuni Project, with the appropriate mailing list contact email address.

Note, _do not_ remove the `Packager` tag under the debbuild vendor conditional, as it's required for Debian packages.

File: client/debian/apt-spacewalk/apt-spacewalk.spec [hunk 2]
```diff
@@ -0,0 +1,150 @@
+%{!?__python2:%global __python2 /usr/bin/python2}
```
Comment (Conan-Kudo) [inline client/debian/apt-spacewalk/apt-spacewalk.spec]: I imagine this package needs to be SUSE-ified a bit (splitting the changelog out, adding the copyright header, etc.)...

File: client/rhel/mgr-daemon/mgr-daemon.spec [hunk 3]
```diff
@@ -28,21 +35,25 @@ Version:        4.0.1
...
 License:        GPL-2.0-only
+%if %{_vendor} == "debbuild"
+Group:          utils
+Packager:       Spacewalk Project <spacewalk-devel@redhat.com>
+%else
 Group:          System Environment/Base
+%endif
 Source0:        spacewalksd-%{version}.tar.gz
 Source1:        %{name}-rpmlintrc
 URL:            https://fedorahosted.org/spacewalk
```
Comment (Conan-Kudo) [inline client/rhel/mgr-daemon/mgr-daemon.spec]: Spacewalk URL is wrong, and should be fixed across the board.

File: client/rhel/mgr-daemon/mgr-daemon.spec [hunk 4]
```diff
@@ -28,21 +35,25 @@ Version:        4.0.1
...
 BuildRoot:      %{_tmppath}/%{name}-%{version}-build
 
-BuildRequires:  gettext
+%if %{_vendor} != "debbuild"
+
 # 5.0.37.2 was last version+1 of spacewalksd before renaming to mgr-daemon
```
Comment (Conan-Kudo) [inline client/rhel/mgr-daemon/mgr-daemon.spec]: Uhh, what?

File: client/debian/apt-spacewalk/apt-spacewalk.spec (hunk 1 above)
Comment (mcalmer) [reply client/debian/apt-spacewalk/apt-spacewalk.spec]: Correct, but we need a short discussion about which address to use. @juliogonzalez - do you have an idea?

File: client/debian/apt-spacewalk/apt-spacewalk.spec (hunk 2 above)
Comment (mcalmer) [reply client/debian/apt-spacewalk/apt-spacewalk.spec]: yes, as soon as we want to build it. For now it is ok. We just pulled in what spacewalk has.

File: client/rhel/mgr-daemon/mgr-daemon.spec (hunk 3 above)
Comment (mcalmer) [reply client/rhel/mgr-daemon/mgr-daemon.spec]: done

File: client/rhel/mgr-daemon/mgr-daemon.spec (hunk 4 above)
Comment (mcalmer) [reply client/rhel/mgr-daemon/mgr-daemon.spec]: The comment refer to the Provides/Obsoletes statements below.
We renamed that package and by this we changed the version to a lower version then the old
package had.

mgr-daemon 4.0.1 obsolete rhnsd   < 5.0.37.2

File: client/debian/apt-spacewalk/apt-spacewalk.spec (hunk 1 above)
Comment (juliogonzalez) [reply client/debian/apt-spacewalk/apt-spacewalk.spec]: I just requested uyuni-devel@opensuse.org, meant for cases such as this one and for development discussions, so you can use it for this PR.

I will send an email to uyuni-announce as soon as it's ready.

**Pull request merged**